core/src/fr/hammons/slinc/modules/LinkageTools.scala

```diff
@@ -80,14 +87,16 @@ object LinkageTools:
           load(cachedFile.cachePath)
 
         case Dependency.LibraryResource(path, false) =>
-          val resolvedPath = potentialArchMarks.view
-            .map(archMark => s"${path}_$archMark$libSuffix")
+          val candidates =
+            resolveCandidates(path, libSuffixCandidates, potentialArchMarks)
+          val resolvedPath = candidates
             .find(path =>
               getClass().getResource(s"$resourcesLocation/$path") != null
             )
             .getOrElse(
               throw Error(
-                s"No library resource found for $arch $os with path like $path"
+                s"No library resource found for $arch $os. Make sure one of ${candidates
```

Comment:
If library is not found, it shows the message like this.

```
Caused by: java.lang.Error: No library resource found for AArch64 Darwin. Make sure one of [test_arm64.dylib, test_aarch64.dylib, test_arm64.so, test_aarch64.so, test.dylib, test.so] is located in resources/native.
```

Reply:
I think this annotation shouldn't search for test.so or test.dylib. It's highly unlikely that such a file will work on all platforms and this annotation is meant to aid platform independence. I think if the user wants to use `test.so` they should specify `@NeedsFile("test.so")` and so on.

Also, this helps guard against the kind of weird errors we might get when loading an AArch64 dylib on amd64 because the user didn't mark their lib `test_aarch64.dylib`

Reply:
> I think this annotation shouldn't search for test.so or test.dylib. It's highly unlikely that such a file will work on all platforms and this annotation is meant to aid platform independence. 

That makes sense.